controller/src/main/java/io/pravega/controller/fault/ControllerClusterListener.java

```diff
@@ -82,6 +87,9 @@ protected void startUp() throws Exception {
                         if (eventProcessorsOpt.isPresent() && eventProcessorsOpt.get().isRunning()) {
                             eventProcessorsOpt.get().notifyProcessFailure(host.getHostId());
                         }
+                        if (txnSweeperOpt.isPresent() && txnSweeperOpt.get().isReady()) {
```

Comment:
If this call to Cluster.AddListener were moved until after the block that is currently ending on line 128- 136 below, this case could be avoided all together.